crates/compiler/codegen/src/generator.rs

```diff
@@ -609,10 +597,10 @@ mod tests {
         // With the direct return optimization, the immediate is stored directly
         // to the return slot at [fp - 3], which is offset -3
         let store_imm = &compiled.instructions[0];
-        assert_eq!(store_imm.opcode, 6); // StoreImm opcode
+        assert_eq!(store_imm.opcode, Opcode::StoreImm);
         assert_eq!(
             store_imm.operands,
-            vec![M31::from(42), M31::from(0), M31::from(-3)]
+            [M31::from(42), Zero::zero(), M31::from(-3)]
```

Comment:
```suggestion
            [M31(42), Zero::zero(), M31::from(-3)]
```
Not sure this works for negative though, to test ?

Reply:
It could be applied everywhere in test

Reply:
these are tests value that are known to fit in u32 and reduce clutter but agree with your point

Reply:
yeah, for example you can't do M31(-3)

Reply:
that is what i though, add doubt about this one.
Anyway for others seems ok

Reply:
You can do `-M31(3)` though, same as `M31(M31::P - 3)`